prime-router/src/main/kotlin/azure/BlobAccess.kt

```diff
@@ -336,6 +338,25 @@ class BlobAccess() : Logging {
             return toBlobUrl
         }
 
+        /**
+         * Download all blobs located at [sourceBlobDirectoryUrl] with container info [sourceBlobContainerInfo] to the
+         * [destinationDirectoryUrl].
+         */
+        fun downloadBlobsInDirectoryToLocal(
```

Comment:
It appears this function would flatten the downloaded files into a single directory instead of preserving a potential virtual hierarchy. I think this is fine for the intended use case but something to note.

Reply:
No, it actually creates the directory structure because of how the function `writeByteArrayToFile` works